types/timepicker/index.d.ts

```diff
@@ -0,0 +1,162 @@
+    scrollDefault?: string;
+
+    // Update the input with the currently highlighted time value when the timepicker loses focus.
+    // default: false
+    selectOnBlur?: boolean;
+
+    // Show "24:00" as an option when using 24-hour time format. You must also set timeFormat for this option to work.
+    // default: false
+    show2400?: boolean;
+
+    // Shows the relative time for each item in the dropdown. minTime or durationTime must be set.
+    // default: false
+    showDuration?: boolean;
+
+    // Display a timepicker dropdown when the input fires a particular event. Set to null or an empty array to disable automatic display. Setting should be an array of strings. default: ['focus']
+    showOn?: string[]|null;
+
+    // DEPRECATED: Display a timepicker dropdown when the input gains focus.
+    // default: true
+    showOnFocus?: boolean;
+
+    // The amount of time, in minutes, between each item in the dropdown.
+    // Alternately, you can specify a function to generate steps dynamically.
+    // The function will receive a count integer (0, 1, 2...) and is expected to return a step integer.
+    // default: 30
+    step?: number;
+
+    // When scrolling on the edge of the picker, it prevent parent containers () to scroll. default: false
+    stopScrollPropagation?: boolean;
+
+    // How times should be displayed in the list and input element. Uses PHP's date() formatting syntax.
+    // Characters can be escaped with a preceeding double slash (e.g. H\\hi).
+    // Alternatively, you can specify a function instead of a string, to use completely custom time formatting.
+    // In this case, the format function receives a Date object and is expected to return a formatted time as a string. default: 'g:ia'
+    timeFormat?: string;
+
+    // Highlight the nearest corresponding time option as a value is typed into the form input.
+    // default: true
+    typeaheadHighlight?: boolean;
+
+    // Convert the input to an HTML <SELECT> control.
+    // This is ideal for small screen devices, or if you want to prevent the user from entering arbitrary values.
+    // This option is not compatible with the following options: appendTo, closeOnWindowScroll, disableTouchKeyboard, forceRoundTime, scrollDefault, selectOnBlur, typeAheadHighlight.
+    // default: false
+    useSelect?: boolean;
+
+    // If a time greater than 24 hours (27:30, for example) is entered, apply modolo 24 to create a valid time.
+    // Setting this to false will cause an input of 27:30 to result in a timeFormatError event.
+    // default: true
+    wrapHours?: boolean;
+}
+
+interface JQuery {
+    timepicker(options: JQueryTimepickerOptions): JQuery;
+
+    /** Get the time as an integer, expressed as seconds from 12am. */
+    timepicker(methodName: 'getSecondsFromMidnight'): number;
+
+    /** Get the time using a Javascript Date object, relative to a Date object (default: today's date). */
+    timepicker(methodName: 'timepicker', date?: Date): Date;
```

Comment:
@sheetalkamat according to the jquery plugin 
need to changed in this line
`timepicker(methodName: 'getTime', date?: Date): Date;`
instead of
`timepicker(methodName: 'timepicker', date?: Date): Date;`